.ci/conda-envs/ci-core-py39.txt

```diff
@@ -10,27 +10,24 @@
 #
 #   conda create \
 #     --name test-env \
-#     --file ./.ci/conda-envs/ci-core-py37.txt
+#     --file ./.ci/conda-envs/ci-core-py39.txt
 #
 
 # python
-python=3.7.*
+python=3.9.*
 
 # direct imports
 cffi=1.15.*
-# older versions of Dask are incompatible with pandas>=2.0, but not all conda packages' metadata accurately reflects that
-#
-# ref: https://github.com/microsoft/LightGBM/issues/6030
-dask=2022.2.*
-distributed=2022.2.*
+# Dask & distributed are the first versions that support tornado >=6.2
+# which is required for testing
+dask=2022.12.*
+distributed=2022.12.*
```

Comment:
I'd support updating the minimum for `dask` in the `[dask]` extra to `>=2022.12.0` to match this.

Here:

https://github.com/microsoft/LightGBM/blob/ec5492f8f915f9bab99e52417cff6808dbdb8ddd/python-package/pyproject.toml#L42

That's 2+ years old at this point, which is a LOT of time given the speed of development in Dask.

Reply:
I'll do this in a follow-up PR, let's not grow the size of this PR too much more.